### DISCUSSION TITLE
Disable InteractivePopGestureRecognizer for specific page

### DISCUSSION THREAD

**Discussion by Divyesh-Bhatt** [dotnet/maui#22369] [Ideas]
### Description

In iOS 15 and above we can perform back navigation using swipe (from left edge of the device). I've noticed that on initial page which is not of type of navigation page if we perform this swipe the app is getting crash in iOS. 

To fix this I've found this work around https://learn.microsoft.com/en-us/answers/questions/306452/disable-xamarin-shell-flyout-swipe-gesture

But this disables the swipe for entire app, which is not intended. So how to make it page specific?

### Public API Changes

ShellPage 

### Intended Use-Case

Disable Swipe back gesture for iOS when needed for specific page.

**Comment by anitatrifunovska** [comment]
To disable the swipe back gesture for iOS on a specific page, you can try the following approach. Hope it helps ! 
```
public TestPage( )
{
    InitializeComponent();
    this.Loaded += TestPage_Loaded;
}

private void TestPage_Loaded(object? sender, EventArgs e)
{
#if IOS
    if (Handler is IPlatformViewHandler platformViewHandler)
    {
        var controller = platformViewHandler.ViewController?.NavigationController;
        if (platformViewHandler.ViewController?.NavigationController != null)
        {
            platformViewHandler.ViewController.NavigationController.InteractivePopGestureRecognizer.Enabled = false;
        }
    }
#endif
}

protected override void OnDisappearing()
{
    base.OnDisappearing();
    this.Loaded -= TestPage_Loaded;
}

**Reply by urnotjl** [reply]
> To disable the swipe back gesture for iOS on a specific page, you can try the following approach. Hope it helps !
> 
> ```
> public TestPage( )
> {
>     InitializeComponent();
>     this.Loaded += TestPage_Loaded;
> }
> 
> private void TestPage_Loaded(object? sender, EventArgs e)
> {
> #if IOS
>     if (Handler is IPlatformViewHandler platformViewHandler)
>     {
>         var controller = platformViewHandler.ViewController?.NavigationController;
>         if (platformViewHandler.ViewController?.NavigationController != null)
>         {
>             platformViewHandler.ViewController.NavigationController.InteractivePopGestureRecognizer.Enabled = false;
>         }
>     }
> #endif
> }
> 
> protected override void OnDisappearing()
> {
>     base.OnDisappearing();
>     this.Loaded -= TestPage_Loaded;
> }
> ```

Works like a charm,thanks! 

I thought have to go through https://learn.microsoft.com/en-us/answers/questions/1143556/how-to-disable-the-swipe-back-in-ios-using-net-mau with Shell , saving much efforts ! 

Plus might have to mark as `... InteractivePopGestureRecognizer.Enabled = True;` in `OnDisappearing()` as well, if anyone keen on making rest navigation pages swipeable?

**Reply by anitatrifunovska** [reply]
Yes, I just tested it and you do need to enable it in the `OnDisappearing()` method.

**Reply by cmattes** [reply]
Works great! Thanks!

**Comment by BlueRaja** [comment]
Here is a method that allows you to keep all your iOS code in your iOS project:

```
Microsoft.Maui.Handlers.PageHandler.Mapper.PrependToMapping("MyApp.MyPageHandler", (handler, view) =>
{
    if (view is MyPage page)
    {
        var navigationController = handler?.PlatformView.GetNavigationController();
        if (navigationController != null)
          navigationController.InteractivePopGestureRecognizer.Enabled = false;
    }
});
```

**Reply by beeradmoore** [reply]
Trying this with a basic test app of a main page and a second page, inside a navigationpage. The `navigationController` is never not null. Do you know if something changed since then?

**Reply by BlueRaja** [reply]
@beeradmoore Yeah that method doesn't seem to work anymore in iOS 26. Here is a method I found that does work, but it disables swiping app-wide, rather than for a single page

```
NavigationRenderer.Mapper.AppendToMapping("MyApp.NavigationRenderer", (handler, _) =>
{
    handler.InteractivePopGestureRecognizer.Enabled = false;

    if (OperatingSystem.IsIOSVersionAtLeast(26))
        handler.InteractiveContentPopGestureRecognizer.Enabled = false;
});
```

**Comment by jgold6** [comment]
For anyone running into this on iOS 26, Apple added a property named **InteractiveContentPopGestureRecognizer**  which is what you need to disable when running on iOS 26. 
You still need to disable **InteractivePopGestureRecognizer** for earlier versions of iOS and make sure you do not attempt to set **InteractiveContentPopGestureRecognizer** on iOS < iOS 26 or you will crash. 

```
if (DeviceInfo.Current.Version.Major >= 26)
{
    uiNavigationController.InteractiveContentPopGestureRecognizer.Enabled = false;
}
else
{
    uiNavigationController.InteractivePopGestureRecognizer.Enabled = false;
}
```
